n64.mk

```diff
@@ -99,10 +106,15 @@ $(BUILD_DIR)/%.o: $(SOURCE_DIR)/%.c
 	@echo "    [CC] $<"
 	$(CC) -c $(CFLAGS) -o $@ $<
 
+$(BUILD_DIR)/%.o: $(SOURCE_DIR)/%.cpp 
+	@mkdir -p $(dir $@)
+	@echo "    [CXX] $<"
+	$(CXX) -c $(CXXFLAGS) -o $@ $<
+
 %.elf: $(N64_ROOTDIR)/mips64-elf/lib/libdragon.a $(N64_ROOTDIR)/mips64-elf/lib/libdragonsys.a $(N64_ROOTDIR)/mips64-elf/lib/n64.ld
 	@mkdir -p $(BUILD_DIR)
 	@echo "    [LD] $@"
-	$(LD) -o $@ $(filter-out $(N64_ROOTDIR)/mips64-elf/lib/n64.ld,$^) $(LDFLAGS) -Map=$(BUILD_DIR)/$(notdir $(basename $@)).map
+	$(LD) -o $@ $(filter-out $(N64_ROOTDIR)/mips64-elf/lib/n64.ld,$^) $(LDFLAGS) $(MAPCMD)=$(BUILD_DIR)/$(notdir $(basename $@)).map
```

Comment:
We already accept `LDFLAGS` which we can introduce extra flags, we shouldn't need something extra here. What I mean is we can just do `LDFLAGS+=..` in the consumer side.

Reply:
@anacierdem the issue here is that the argument of `-Map` is not inside the `LDFLAGS` variable and when you use `g++` as a linker you can't just pass `-Map` to g++. It expects the argument to be `-Wl,-Map` so that `g++` knows its not a `g++` argument and it can hand it down to the linker.

Reply:
can't we just switch to always using g++ as linker? Does that create a problem for C programs?

Reply:
I think the g++ driver should only add the C++ standard library to the default library list, and the linker should then remove all references to it (hopefully). 

This also begs the question whether using `-lstdc++` with ld is enough to fix C++ code.

Reply:
I have tried wrapping all linker flags with `-Wl,` but that does not play well with `-lc` next step is trying `-lstdc++` as @rasky has suggested. If that does not also work for some reason I will try linking everything with `g++` providing `-lc` for non-C++ code.

Reply:
I can try that as well if it is not known to cause any issues.